src/pages/instances/InstanceSnapshots.tsx

```diff
@@ -71,46 +113,107 @@ const InstanceSnapshots: FC<Props> = ({ controlTarget, instance }) => {
     };
   });
 
+  const {
+    pageData: pageSnapshots,
+    currentPage,
+    paginate: setCurrentPage,
+  } = usePagination(rows, {
+    itemsPerPage: pageSize,
+    autoResetPage: true,
+  });
+
+  const updateTBodyHeight = () => {
+    const table = document.getElementById("snapshots-table-wrapper");
+    if (!table || table.children.length !== 2 || !instance.snapshots) {
+      return;
+    }
+    const tBody = table.children[1];
+    const above = tBody.getBoundingClientRect().top + 1;
+    const below = 85;
+    const offset = Math.ceil(above + below);
+    const style = `height: calc(100vh - ${offset}px); min-height: calc(100vh - ${offset}px)`;
+    tBody.setAttribute("style", style);
+  };
+  useEventListener("resize", updateTBodyHeight);
+  useEffect(() => {
+    updateTBodyHeight();
+  }, [instance.snapshots, inTabNotification, query, pageSize, currentPage]);
+
   return (
-    <>
-      {controlTarget &&
-        createPortal(
-          <>
-            <CreateSnapshotBtn openSnapshotForm={() => setModalOpen(true)} />
-          </>,
-          controlTarget
-        )}
+    <div className="snapshot-list">
       {isModalOpen && (
         <CreateSnapshotForm
           instance={instance}
           close={() => setModalOpen(false)}
+          onSuccess={onSuccess}
         />
       )}
-      <Row>
-        {instance.snapshots && instance.snapshots.length > 0 ? (
+      {hasSnapshots && (
+        <div className="upper-controls-bar">
+          <div className="search-box-wrapper">
+            <SearchBox
+              className="search-box margin-right"
```

Comment:
This is missing an `id` field. If it is there, a hidden label will be linked to the input. Otherwise, we get a hidden label, that has no connected form element.

Reply:
I was wrong. What I wrote is true for input, select and other form elements. For searchbox we must have a name, but not an id. The component wants to id itself as search, so it shouldn't be set. Something we could fix upstream at one point, so the provided I'd won't break it.

Reply:
Added a `name` to the `SearchBox`.